### PR TITLE
New TMI Interface

### PR DESCRIPTION
**Pull request by stachu540**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Changes Proposed
* TMI package created and refactoring PubSub and Chat it is on `me.philippheuer.twitch4j.tmi`
* Added IRC over WebSocket support (required additional help)
* incomplete Parser (I will work on it)
*  WebSocket Listener - only listening Pings and Pongs. (additional help required for event listening)
* refactoring package, while they are in `me.philippheuer.twitch4j.tmi` package
* both are now WebSocket, and they have connection statement (use `getConnection()` method) - also, they have inside the code enum `Connection` class.
* reformating regex to ASCII characters - http://fiddle.re/fbcndd
* backslashes before @ [at] removed - http://fiddle.re/fbcndd
* shredding coppule code-lines
* all Twitch endpoints are now using Endpoints enum class (for convenience)
* PubSub changes ([explanation](#issuecomment-323611548))

### Additiona Notes
Subs, Whispers, Bits and Commercials can be getted from [PubSub](https://dev.twitch.tv/docs/v5/guides/PubSub/) system. You can also get this topic receiving `USERNOTICE` [twitch tag](https://dev.twitch.tv/docs/v5/guides/irc/#twitch-irc-capability-tags). Better way is using PubSub. Other way is using [Twitch-Specific IRC Capabilities](https://dev.twitch.tv/docs/v5/guides/irc/#twitch-specific-irc-capabilities). Everything is on Twitch Documentation. 

EDIT: Travis will be dropped out some errors cause classes is incomplete. Today, im tired and confused. I will make rest job, tomorrow.
EDIT2: Founded issue when I try getting user. RestClient returns `no client id specified`.

### CODE REVIEW

**Comment by stachu540** [comment]
Whoops. Wrong branch. 😝

**Comment by stachu540** [comment]
Still not done yet. Im still testing and adding some missing parts.

File: build.gradle [hunk 1]
```diff
@@ -50,8 +50,6 @@ dependencies {
 	// WebSocket Client for Twitch PubSub
 	compile group: 'com.neovisionaries', 			name: 'nv-websocket-client',	version: '2.3'
 
-	// IRC Client [https://github.com/KittehOrg/KittehIRCClientLib]
-	compile group: 'org.kitteh.irc', 				name: 'client-lib', 			version: '3.2.0'
 	// - Transitive Dependency Conflict: Ratpack <-> Kitteh IRC Client
```
Comment (stachu540) [inline build.gradle]: Now you can use Ratpack if IRC Client has been removed

**Comment by stachu540** [comment]
Changes improved. Watch @ first post.

If you wish add somethin', go ahead. My part of job is done. but still needs changes at IRCParser. 🤔 
Ready for reviewing.

**Comment by stachu540** [comment]
### Explanation workout PubSub changes:
Using enum class `PubSubTopics`, we can define which topics are be listening.
to listening some channel we can use method `listenChannel()` with channel name and Topics.

For example we need listening bits from channel `summit1g` (it is example) then we using:
```java
twitchClient.getTMI().getPubSub().listenChannel("summit1g", PubSubTopics.BITS);
```

List supported topics with required scopes finds [here](https://dev.twitch.tv/docs/v5/guides/PubSub/#subscribing-to-topics)

**Comment by stachu540** [comment]
I think, that's all. If I something missing just name it.

**Comment by PhilippHeuer** [comment]
I tested irc today and it works well, the messages are the same as in irc before, so we should be able to use the same code to parse them.

I just renamed a few methods & variables and fixed a few issues.

File: build.gradle (hunk 1 above)
Comment (PhilippHeuer) [inline build.gradle]: Nice, if KittehIRC is gone we can remove the entire `Transitive Dependency Conflict: Ratpack <-> Kitteh IRC Client` part (netty dependencies) from the build.gradle.

**Comment by stachu540** [comment]
Just add new branch. I will change merge branch and you will merge it.

**Pull request merged**